### PR TITLE
Fix remaining blackbox test issues

### PR DESCRIPTION
**Pull request by paescuj**
## Scope

- Timezone tests were failing due to the change from #22234, the clear cache request has now been adjusted
- Configure max number of forks, reducing pressure when running tests, making it succeed locally again 
- Clean-up no longer required workarounds / logs for flaky tests
- Fix hanging of the blackbox test, by removing leftover test in sequential list
- Update Vitest to [v1.6.0](https://github.com/vitest-dev/vitest/releases/tag/v1.6.0), which comes with performance improvement

## Potential Risks / Drawbacks

None

## Review Notes / Questions

Successful run: https://github.com/directus/directus/actions/runs/9549029952/job/26317643526
- There are still two permission tests failing which will be addressed in #22773

---

Related to #20502

### CODE REVIEW

File: .github/workflows/blackbox.yml
Comment (licitdev) [inline .github/workflows/blackbox.yml:1]: Let's revert this change if the PR is ready for review ❤️

**Comment by rijkvanzanten** [comment]
@licitdev You're the one with the most experience in this setup. Mind taking a look and merging if everything looks OK? 🙂

**Review by licitdev — COMMENTED** [review]
Should we update the `TEST_NO_CACHE` port to use `getPort()` or add to the config as a `PORT_NO_CACHE`? Or both? 🤔 

https://github.com/directus/directus/blob/752196d7335b88652f878f5cb4fceb64f493923d/tests/blackbox/common/config.ts#L324
https://github.com/directus/directus/blob/752196d7335b88652f878f5cb4fceb64f493923d/tests/blackbox/setup/setup.ts#L82

**Review by licitdev — APPROVED** [review]
LGTM 🚀 

Note: Changes to implement `getPort` across all port usages will be patched separately in the future.
